plone/namedfile/utils/jpeg_utils.py

```diff
@@ -10,17 +10,16 @@
 
 
 def process_jpeg(data):
+    content_type = None
+    w = -1
+    h = -1
     size = len(data)
-    content_type, w, h = None, -1, -1
 
     if (size >= 2) and data.startswith('\377\330'):  # handle JPEGs
-        content_type = 'image/jpeg'
```

Comment:
This is the point where it should be set, as those two bytes indicate the content_type.